File: android/AndroidJetpackBook/Chapter_08/SimpleNavigationDemo/app/src/main/java/com/example/simplenavigationdemo/FragmentB.java
package com.example.simplenavigationdemo;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

public class FragmentB extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_b, container, false);
        view.findViewById(R.id.fragment_jump_to_next_btn).setOnClickListener(v ->
                Navigation.findNavController(view).navigate(R.id.fragmentC)
        );

        return view;
    }
}

File: android/AndroidJetpackBook/Chapter_08/SimpleNavigationDemoWithKTX/app/src/main/java/com/example/simplenavigationdemowithktx/FragmentB.kt
package com.example.simplenavigationdemowithktx

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation
import com.example.simplenavigationdemowithktx.R

class FragmentB : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_b, container, false)
        view.findViewById<Button>(R.id.fragment_jump_to_next_btn)
            .setOnClickListener {
                Navigation.findNavController(view).navigate(NavGraph.Dest.fragment_c)
            }
        return view;
    }
}